sycl/source/detail/scheduler/commands.cpp

```diff
@@ -110,8 +110,12 @@ static void applyFuncOnFilteredArgs(
           ++NextTrueIndex;
       LastIndex = Arg.MIndex;
 
-      if (EliminatedArgMask[Arg.MIndex])
+      if (EliminatedArgMask[Arg.MIndex]) {
+        if (Arg.MType == kernel_param_kind_t::kind_accessor)
```

Comment:
This runs contrary to what I would expect from a function called `applyFuncOnFilteredArgs`. Moving out a part of this filtering to the function object passed by the callee seems wrong to me (especially considering that this filtering is missing from the call site at `ExecCGCommand::emitInstrumentationData` which would result in invalid xpti metadata).